cpr/sse.cpp

```diff
@@ -0,0 +1,115 @@
+#include "cpr/sse.h"
+
+#include <algorithm>
+#include <cctype>
+#include <charconv>
+
+namespace cpr {
+
+bool ServerSentEventParser::parse(std::string_view data, const std::function<bool(ServerSentEvent&&)>& callback) {
+    // Append incoming data to buffer
+    buffer_.append(data);
+
+    // Process complete lines
+    size_t pos = 0;
+    while ((pos = buffer_.find('\n')) != std::string::npos) {
+        std::string line = buffer_.substr(0, pos);
+        buffer_.erase(0, pos + 1);
+
+        // Remove trailing \r if present (handles both \n and \r\n)
+        if (!line.empty() && line.back() == '\r') {
+            line.pop_back();
+        }
+
+        if (!processLine(line, callback)) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void ServerSentEventParser::reset() {
+    buffer_.clear();
+    current_event_ = ServerSentEvent();
+}
+
+bool ServerSentEventParser::processLine(const std::string& line, const std::function<bool(ServerSentEvent&&)>& callback) {
+    // Empty line means end of event
+    if (line.empty()) {
+        return dispatchEvent(callback);
+    }
+
+    // Lines starting with ':' are comments, ignore them
+    if (line[0] == ':') {
+        return true;
+    }
+
+    // Find the colon separator
+    size_t colon_pos = line.find(':');
+
+    std::string field;
+    std::string value;
+
+    if (colon_pos == std::string::npos) {
+        // No colon, entire line is the field name
+        field = line;
+        value = "";
+    } else {
+        field = line.substr(0, colon_pos);
+        // Skip the colon and optional leading space
+        size_t value_start = colon_pos + 1;
+        if (value_start < line.size() && line[value_start] == ' ') {
```

Comment:
Does it make sense to trim the string left and right in case there are multiple white space characters?